packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -189,11 +189,39 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
         const widget = this.shell.currentWidget;
         if (widget instanceof TerminalWidget) {
             this.setCurrentTerminal(widget);
+            this.setLastUsedTerminal(widget);
```

Comment:
Should not it go into `setCurrentTerminal`?

Reply:
Two differences between `setCurrentTerminal()` and `setLastUsedTerminal()`:

1) `setCurrentTerminal()` can be called with `undefined`, i.e., `this.setCurrentTerminal(undefined)`, meaning the "current terminal" is not any of the opened terminal widgets.
`setLastUsedTerminal()` won't be called with `undefined`.

2) `setLastUsedTerminal()` can be called with terminal widget that does not have focus, while `setCurrentTerminal()` (except for `undefined`) is always called with a widget that has the focus.

So i did not put the `setCurrentTerminal` calls into `setCurrentTerminal`

Reply:
ok